test/rekt/features/broker/feature.go

```diff
@@ -200,3 +203,208 @@ func brokerRedeliveryDropN(retryNum int32, dropNum uint) *feature.Feature {
+
+	source := feature.MakeRandomK8sName("source")
+	sink := feature.MakeRandomK8sName("sink")
+	via := feature.MakeRandomK8sName("via")
+
+	eventSource := "source1"
+	eventType := "type1"
+	eventBody := `{"msg":"test msg"}`
+	event := cloudevents.NewEvent()
+	event.SetID(uuid.New().String())
+	event.SetType(eventType)
+	event.SetSource(eventSource)
+	event.SetData(cloudevents.ApplicationJSON, []byte(eventBody))
+
+	//Install the broker
+	brokerName := feature.MakeRandomK8sName("broker")
+	f.Setup("install broker", broker.Install(brokerName, broker.WithEnvConfig()...))
+	f.Requirement("broker is ready", broker.IsReady(brokerName))
+	f.Requirement("broker is addressable", broker.IsAddressable(brokerName))
+
+	f.Setup("install sink", eventshub.Install(sink, eventshub.StartReceiver))
+
+	// Point the Trigger subscriber to the sink svc.
+	cfg := []manifest.CfgFn{
+		trigger.WithSubscriber(nil, "http://fake.svc.cluster.local"),
+		trigger.WithDeadLetterSink(svc.AsKReference(sink), ""),
+	}
+
+	// Install the trigger
+	f.Setup("install trigger", trigger.Install(via, brokerName, cfg...))
+	f.Setup("trigger goes ready", trigger.IsReady(via))
+
+	f.Requirement("install source", eventshub.Install(
+		source,
+		eventshub.StartSenderToResource(broker.GVR(), brokerName),
+		eventshub.InputEvent(event),
+	))
+
+	f.Assert("Receives dls extensions when subscriber is unreachable",
+		eventasssert.OnStore(sink).
+			MatchEvent(
+				test.HasExtension("knativeerrordest", "http://fake.svc.cluster.local"),
+			).
+			AtLeast(1),
+	)
+	return f
+}
+
+func brokerSubscriberErrorNodata() *feature.Feature {
+	f := feature.NewFeatureNamed("Broker Subscriber Error Nodata")
+
+	source := feature.MakeRandomK8sName("source")
+	sink := feature.MakeRandomK8sName("sink")
+	failer := feature.MakeRandomK8sName("failer")
+	via := feature.MakeRandomK8sName("via")
+
+	eventSource := "source1"
+	eventType := "type1"
+	eventBody := `{"msg":"test msg"}`
+	event := cloudevents.NewEvent()
+	event.SetID(uuid.New().String())
+	event.SetType(eventType)
+	event.SetSource(eventSource)
+	event.SetData(cloudevents.ApplicationJSON, []byte(eventBody))
+
+	//Install the broker
+	brokerName := feature.MakeRandomK8sName("broker")
+	f.Setup("install broker", broker.Install(brokerName, broker.WithEnvConfig()...))
+	f.Requirement("broker is ready", broker.IsReady(brokerName))
+	f.Requirement("broker is addressable", broker.IsAddressable(brokerName))
+
+	f.Setup("install sink", eventshub.Install(sink, eventshub.StartReceiver))
+
+	f.Setup("install failing receiver", eventshub.Install(failer,
+		eventshub.StartReceiver,
+		eventshub.DropFirstN(1),
+		eventshub.DropEventsResponseCode(422),
+	))
+
+	// Point the Trigger subscriber to the sink svc.
+	cfg := []manifest.CfgFn{
+		trigger.WithSubscriber(svc.AsKReference(failer), ""),
+		trigger.WithDeadLetterSink(svc.AsKReference(sink), ""),
+	}
+
+	// Install the trigger
+	f.Setup("install trigger", trigger.Install(via, brokerName, cfg...))
```

Comment:
similar here 
https://github.com/knative-sandbox/eventing-kafka-broker/blob/main/test/e2e_new/dls_extensions_test.go#L146-L151

for the `SubscriberReturnedErrorNoData`

Reply:
Ok, I'll format this place